src/Meziantou.Analyzer/Rules/DoNotUseElseWhenIfBlockJumpsAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Meziantou.Analyzer.Rules
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class DoNotUseElseWhenIfBlockJumpsAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor s_rule = new DiagnosticDescriptor(
+            RuleIdentifiers.DoNotUseElseWhenIfBlockJumps,
+            title: "Do not use 'else' when 'if' block jumps",
+            messageFormat: "Do not use 'else' when 'if' block jumps",
+            RuleCategories.Style,
+            DiagnosticSeverity.Info,
+            isEnabledByDefault: true,
+            description: "The 'if' block ends with a jump statement (break, continue, goto, return, throw, yield). Using 'else' is redundant and needlessly maintains a higher nesting level.",
+            helpLinkUri: RuleIdentifiers.GetHelpUri(RuleIdentifiers.DoNotUseElseWhenIfBlockJumps));
+
+        private static readonly Type[] s_jumpStatementSyntaxTypes =
+        {
+            typeof(BreakStatementSyntax),
+            typeof(ContinueStatementSyntax),
+            typeof(GotoStatementSyntax),
+            typeof(ReturnStatementSyntax),
+            typeof(ThrowStatementSyntax),
+            typeof(YieldStatementSyntax),
+        };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(AnalyzeElseClause, SyntaxKind.ElseClause);
+        }
+
+        private static void AnalyzeElseClause(SyntaxNodeAnalysisContext context)
+        {
+            var elseClauseSyntax = (ElseClauseSyntax)context.Node;
+
+            var ifStatementSyntax = elseClauseSyntax.Parent;
+            var ifStatementNodes = ifStatementSyntax.ChildNodes();
```

Comment:
When working with the syntax tree, you should be very careful. Most of the time, the code you analyze is invalid (it doesn't even compile). So, be sure to check the parent exists and is actually an if statement (`IfStatementSyntax`). Then use the `Statement` property instead of `var statementOrBlock = ifStatementNodes.Skip(1).Take(1).First();` (also check it is not null)

Reply:
You're right. I mostly focused on the happy path.
I now added checks to make the code more robust.